### ISSUE TITLE
Gradient disconnected after Multiprocessing pool(starmap)

### ISSUE THREAD

**Issue by ahyunSeo** [pytorch/pytorch#12202]
## ❓ Questions and Help

Hello,
I'm using pytorch 0.4.1 with python 3.6, Anaconda, Ubuntu 16.04, Titan XP.
I have a problem with multiprocessing.
**Basically I have to do sample-specific "non-batchable" operations within the batch,
so to make it faster I tried to use multiprocessing.**
The operation consists of computing losses in sequential manner,
but each of the samples have different length of the sequences.
Anyway,
I used pool and starmap.
As an argument of the starmap I tried zip of torch.chunk and list of label_path.
**I checked that until the torch chunk stage, the gradient works fine, 
but after it gets into the sub-process which is get_reward_loss, the gradient is disconnected.**

> 1. How can I connect it?
> 2. It it is impossible, is there any other way to do for-loop within the batch efficiently? (I know about Data-parallel)

Here's the simplified snippet of my code.

```
def get_reward_loss(feat, path):
    print(feat.requires_grad)   # Already False
    # Do some operations to make input_feat
    # Using path, get label and do some processing to get target
    loss = criterion(input_feat, target)
    return loss

train_loader = DataLoader(MNISTToy(data_path, data_transform), batch_size=1, num_workers=1)

optimizer = optim.Adam(net.parameters(), lr=0.01)
t0 = datetime.now()

if __name__ == '__main__':
    set_start_method('spawn', force=True)
    pool = Pool(processes=4)
    for idx, (input, label_path) in enumerate(train_loader):
        input = input.to(device)
        optimizer.zero_grad()
        output = net(input)
        batch_loss = pool.starmap(get_reward_loss, zip(torch.chunk(output, batch_size, dim=0), label_path))
        loss = # sum of batch_loss(this is list)
        loss.backward()
        optimizer.step()

```

**Comment by zou3519** [comment]
This is expected. A tensor without a .grad field will have independent .grad fields after being sent to different processes.

However, a tensor with an existing .grad field will have its .grad field shared among different processes. You should do something like
```
# before sending tensor to a different process
tensor.grad = torch.zeros_like(tensor)
# send tensor to a different process
```
and then the gradients will all be connected.
